indico/modules/events/registration/client/js/form/fields/TimetableSessions.jsx

```diff
@@ -0,0 +1,297 @@
+            </Singular>
+            <Plural>
+              (<Param name="selectedBlocks" value={selectedBlocks} /> blocks selected)
+            </Plural>
+          </PluralTranslate>
+        </span>
+      </AccordionTitle>
+      <AccordionContent active={isExpanded}>
+        {data.map(({startTime, endTime, fullTitle, id}) => (
+          <dd className="grouped-fields" key={id}>
+            <Checkbox
+              value={id}
+              onChange={handleValueChange}
+              checked={value.includes(id)}
+              label={`${startTime} - ${endTime} - ${fullTitle}`}
+            />
+          </dd>
+        ))}
+      </AccordionContent>
+    </>
+  );
+}
+
+SessionBlockHeader.propTypes = {
+  value: PropTypes.arrayOf(PropTypes.number).isRequired,
+  index: PropTypes.number.isRequired,
+  data: sessionDataSchema.isRequired,
+  label: PropTypes.string.isRequired,
+  isExpanded: PropTypes.bool.isRequired,
+  handleValueChange: PropTypes.func.isRequired,
+  onChange: PropTypes.func.isRequired,
+  onClick: PropTypes.func.isRequired,
+};
+
+export default function TimetableSessionsInput({
+  htmlId,
+  htmlName,
+  disabled,
+  isRequired,
+  display,
+  minimum,
+  maximum,
+}) {
+  const {eventId} = useSelector(getStaticData);
+  const {data: sessionData} = useIndicoAxios(
+    {url: timeTableURL({event_id: eventId})},
+    {camelize: true}
+  );
+
+  const minMsg = PluralTranslate.string(
+    'Please select at least {minimum} session block',
+    'Please select at least {minimum} session blocks',
+    minimum,
+    {minimum}
+  );
+  const maxMsg = PluralTranslate.string(
+    'Please select no more than {maximum} session block',
+    'Please select no more than {maximum} session blocks',
+    maximum,
+    {maximum}
+  );
+
+  return (
+    <FinalField
+      id={htmlId}
+      name={htmlName}
+      component={TimetableSessionsComponent}
+      disabled={disabled}
+      required={isRequired}
+      display={display}
+      minimum={minimum}
+      maximum={maximum}
+      sessionData={sessionData}
+      validate={value => {
+        if (minimum !== 0 && value.length < minimum) {
+          return minMsg;
+        } else if (maximum !== 0 && value.length > maximum) {
+          return maxMsg;
+        }
+      }}
+    />
+  );
+}
+
+TimetableSessionsInput.propTypes = {
+  htmlId: PropTypes.string.isRequired,
+  htmlName: PropTypes.string.isRequired,
+  isRequired: PropTypes.bool.isRequired,
+  disabled: PropTypes.bool,
+  display: PropTypes.string,
+  minimum: PropTypes.number,
+  maximum: PropTypes.number,
+};
+
+TimetableSessionsInput.defaultProps = {
+  disabled: false,
+  display: 'expanded',
+  minimum: 0,
+  maximum: 0,
+};
+
+export function TimetableSessionsSettings() {
+  return (
+    <>
+      <Form.Group inline>
+        <label>{Translate.string('Default display for sessions')}</label>
+        <FinalRadio name="display" label={Translate.string('Expanded dropdown')} value="expanded" />
```

Comment:
Why not use a Dropdown to select the display mode like we do for the single choice display mode? I think it looks much better...

![image](https://github.com/indico/indico/assets/179599/bedd218b-77f4-4b00-9d3d-cf4787abc691)

![image](https://github.com/indico/indico/assets/179599/aa4122be-1f27-4777-80b5-f7e6ec57524e)

It's even more important in languages where those labels may be even longer!

Reply:
For 2 options only I don't feel it's correct to use a dropdown, but if you have a strong opinion I can do it, I simply feel it's better as it is

Reply:
It'd be more consistent with what we already have, and generally side-by-side checkboxes are considered poor UX because besides the spacing it's sometimes hard to see which label belongs to which checkbox.